DOI.js

```diff
@@ -71,67 +160,175 @@ function getDOIs(doc, url) {
 	return Array.from(new Set(fromURL ? [fromURL, ...fromDocument] : fromDocument));
 }
 
-function getDOIFromURL(url) {
-	// Split on # and ?, so that we don't allow DOIs to contain those characters
-	// but do allow finding DOIs on either side of them (e.g. a DOI in the URL hash)
-	let urlParts = url.split(/[#?]/);
-	for (let urlPart of urlParts) {
-		let match = DOIre.exec(urlPart);
-		if (match) {
-			// Only return a single DOI from the URL
-			return match[0];
+// NOTE: We should case-normalize according to the DOI spec, which says the
+// registry uses all uppercase internally. We follow this convention.
```

Comment:
I don't think we want to capitalize all DOIs. Regardless of how DOIs are stored in the registry, different publishers have different case conventions, and I think we can assume that they're rendered in the preferred manner, which tends to be much more commonly lowercase.

The place where we want to normalize is when doing comparisons (e.g., a recent bug in the app regarding retraction checks).

Reply:
Oh, right. But then is there any point in normalizing in this translator?

Reply:
Ah, got it — that makes sense.